### PR TITLE
Support for pymongo 3

### LINKED ISSUE #207: PyMongo 3 compatibility
"auto_start_request" option is gone, "max_pool_size" renamed to "maxPoolSize", "safe" is gone, "_connect" is renamed "connect".

### PR DESCRIPTION
**Pull request by markunsworth**
Support for Pymongo 3.0.
- Removed use of the `fields` keyword argument when calling find, as this has changed to `projection` in pymongo 3.0. Use positional argument instead.
- Have removed the use  of `safe` and `slave_okay` keyword arguments as these are no longer used and have been deprecated since 2.4

Fix the broken tests.
 - Have changed the `DatabaseOperation.sqlflush` method to now check if a collection is capped before trying to remove the data
 - Removed tests that are no longer relevant.

Fixes #207 and #209

### CODE REVIEW

**Comment by markunsworth** [comment]
Fixes issues mentioned in #210

**Comment by aburgel** [comment]
I think this needs to be rebased since I merged in your safe writes PR.

**Comment by aburgel** [comment]
Can you also pull in some of the documentation updates from #208?

**Comment by markunsworth** [comment]
Ok - will merge those into this branch.

The main question here is, should this still be backwards compatible with pymongo < 3.0. I can probably add some  additional code to try and support older versions of pymongo, but I don't think it'll be pretty

**Comment by aburgel** [comment]
@ajdavis mentioned that pymongo 2.9 will be compatible with both 2.8 and 3.0, so can we standardize on that?

If its too much work, I'm happy to cut a new version and require 3.0.

**Comment by markunsworth** [comment]
We can require pymongo >= 2.9, but i don't think some of the other changes (e.g. removing MongoReplicaSetClient in favour of MongoClient for connecting to replica sets) are being backported. @ajdavis might be able to clarify on that. 

Pymongo 3 still supports the old crud API (find, update, remove) so I think anyone using it directly will still be able to do pretty much everything they are doing with 2.x

**Comment by ajdavis** [comment]
MongoReplicaSetClient lives on in 3.0 as a synonym for MongoClient, so you
can continue to use it.

**Comment by markunsworth** [comment]
So this is now failing on Django 1.6 due to the use of the `atomic()` context manager in the [sql flush method](https://github.com/django-nonrel/django/blob/nonrel-1.6/django/core/management/commands/flush.py#L66)

Given that mongodb (and other non rel databases AFAIK) don't support transactions, I was wondering if we should patch this to not use `atomic()` if the database feature `atomic_transactions` is set to false (it is by default but we can override in djangotoolbox or mongodb-engine).

**Comment by markunsworth** [comment]
There seems to be an issue since moving to use the nonrel-1.6 branch from nonrel-1.6-beta use to the use of `transaction.atomic()` in the flush command.

Added a workaround of setting `DatabaseFeatures.can_rollback_ddl = True` which works for the mongodb-engine tests, but not for the djangotoolbox ones.

**Comment by markunsworth** [comment]
ok - we now have backwards compatibility with 2.x

Have also added more configurations to the environment matrix to test against the latest 2.x and 3.x versions

File: .travis.yml [hunk 1]
```diff
@@ -12,18 +12,28 @@ python:
...
-  - DJANGO_VERSION=1.5
-  - DJANGO_VERSION=1.6
+  - DJANGO_VERSION=1.4 PYMONGO_VERSION=2.8.1
+  - DJANGO_VERSION=1.4 PYMONGO_VERSION=3.0.2
+  - DJANGO_VERSION=1.5 PYMONGO_VERSION=2.8.1
+  - DJANGO_VERSION=1.5 PYMONGO_VERSION=3.0.2
+  - DJANGO_VERSION=1.6 PYMONGO_VERSION=2.8.1
+  - DJANGO_VERSION=1.6 PYMONGO_VERSION=3.0.2
+  - DJANGO_VERSION=1.7 PYMONGO_VERSION=2.8.1
+  - DJANGO_VERSION=1.7 PYMONGO_VERSION=3.0.2
```
Comment (aburgel) [inline .travis.yml]: does django 1.7 actually work?

File: django_mongodb_engine/base.py [hunk 2]
```diff
@@ -69,6 +74,8 @@ def sql_flush(self, style, tables, sequence_list, allow_cascade=False):
             options = collection.options()
 
             if not options.get('capped', False):
+
+                # TODO:Not backwards compatible
```
Comment (aburgel) [inline django_mongodb_engine/base.py]: what's this?

File: requirements.txt [hunk 3]
```diff
@@ -1,4 +1,6 @@
-pymongo<3.0
+
+pymongo>=3.0
```
Comment (aburgel) [inline requirements.txt]: should this be changed to >=2.9?

File: tests/lookup/tests.py [hunk 4]
```diff
@@ -18,6 +18,11 @@
 class LookupTests(TestCase):
 
     def setUp(self):
+
+        Author.objects.all().delete()
+        Article.objects.all().delete()
+        Tag.objects.all().delete()
```
Comment (aburgel) [inline tests/lookup/tests.py]: this looks the same as the following 3 lines

File: tests/mongodb/tests.py [hunk 5]
```diff
@@ -3,16 +3,16 @@
 
 from django.core.management import call_command
 from django.contrib.sites.models import Site
-from django.db import connection, connections
+from django.db import connection
 from django.db.utils import DatabaseError, IntegrityError
 from django.db.models import Q
-
+from encodings.big5 import codec
```
Comment (aburgel) [inline tests/mongodb/tests.py]: you're importing this but not using it.

**Comment by aburgel** [comment]
does any of this depend on django-nonrel/django#22?

File: .travis.yml (hunk 1 above)
Comment (markunsworth) [inline .travis.yml]: Not yet (am planning to look at that next). I've added it to the allowable failures for now

File: django_mongodb_engine/base.py (hunk 2 above)
Comment (markunsworth) [inline django_mongodb_engine/base.py]: A comment i'd forgotten to remove after making this backwards compatible. Was using the new pymongo 3 deleteMany

**Comment by markunsworth** [comment]
@aburgel No, django-nonrel/django#22 isn't necessary for this to work.

File: tests/lookup/tests.py (hunk 4 above)
Comment (markunsworth) [inline tests/lookup/tests.py]: ah, thought i'd removed all of these. Thanks

**Pull request merged**